lang/java/avro/src/test/java/org/apache/avro/reflect/TestReflectData.java

```diff
@@ -130,4 +130,34 @@ private void validateSchema(Meta meta) {
       assertEquals("Invalid field " + field.name(), field.defaultVal(), testCases.get(field.name()));
     }
   }
+
+  public class Definition {
+    public Map<String, String> tokens;
+  }
+
+  @Test
+  public void testInnerClasses() {
+    testStaticInnerClasses();
+    testNonStaticInnerClasses();
+  }
+
+  public void testNonStaticInnerClasses() {
```

Comment:
I'd prefer to use the annotation: `@Test(expected = AvroTypeException.class)`
What do you think?

Reply:
@Fokko I agree! :)